backend/src/main/java/com/google/sps/data/MatchPreference.java

```diff
@@ -42,16 +42,18 @@ public static MatchPreference forIntValue(int value) {
     throw new IllegalStateException("Unknown enum int value.");
   }
 
+  /** Parse match preference string to MatchPreference */
   public static MatchPreference forStringValue(String value) {
-    switch (value.toLowerCase()) {
+    String lowerCasePreference = value.toLowerCase();
+    switch (lowerCasePreference) {
       case "different":
         return MatchPreference.DIFFERENT;
       case "any":
         return MatchPreference.ANY;
       case "similar":
         return MatchPreference.SIMILAR;
       default:
-        throw new IllegalStateException("Unknown enum String value.");
+        throw new IllegalStateException("Unknown enum value.");
```

Comment:
while we're here, let's concatenate `value` to the exception reason. That way if we ever get an unexpected value we know what it is and it'll be easier to debug.